WC_Layers/OpenMC_PhotonTransport.py

```diff
@@ -0,0 +1,136 @@
+import openmc
+import numpy as np
+import yaml
+import argparse
+from Source_Mesh_Reader import extract_source_data
+from TwoLayers_Materials import alara_element_densities, make_materials
+from TwoLayers_Geometry import make_spherical_shells
+
+def make_source(bounds, cells, mesh_file, source_mesh_index, esd):
+    '''
+    Creates a list of OpenMC sources, complete with the relevant space and energy distributions
+    
+    inputs:
+        bounds : iterable of photon energy bounds (float)
+        cells: list of OpenMC Cell objects
+        mesh_file: .h5/.h5m mesh onto which photon source will be distributed
+        source_mesh_index: index specifying the photon source from which data is extracted
+        
+    output:
+        source_list: list of OpenMC independent sources
+        total_mesh: OpenMC Unstructured Mesh object
+    '''
+
+    source_list = []
+    total_mesh = openmc.UnstructuredMesh(mesh_file, library='moab')
+    for index, (lower_bound, upper_bound) in enumerate(zip(bounds[:-1],bounds[1:])):
+        mesh_dist = openmc.stats.MeshSpatial(total_mesh, strengths=esd[source_mesh_index][:,index], volume_normalized=False)
+        energy_dist = openmc.stats.Uniform(a=lower_bound, b=upper_bound)
+        source_list.append(openmc.IndependentSource(space=mesh_dist, energy=energy_dist, strength=np.sum(esd[source_mesh_index][:, index]), particle='photon', domains=cells))
+    return source_list, total_mesh
+
+def make_tallies(total_mesh, tallied_cells, coeff_geom):
+    '''
+    Creates tallies and assigns energy, spatial, and particle filters.
+    
+    inputs: 
+        total_mesh: OpenMC unstructured mesh object
+        tallied_cells: OpenMC Cell/iterable of OpenMC Cell objects/iterable of Cell ID #
+        
+    outputs:
+        OpenMC Tallies object
+    '''
+    particle_filter = openmc.ParticleFilter('photon')
+    total_filter = openmc.MeshFilter(total_mesh)
+    
+    photon_tally = openmc.Tally(tally_id=1, name="Photon tally")
+    photon_tally.scores = ['flux', 'elastic', 'absorption']
+    
+    cell_filter = openmc.CellFilter(tallied_cells)
+    photon_tally.filters = [cell_filter, total_filter, particle_filter]
+    
+    # Obtain coefficients to calculate effective dose
+    dose_energy, dose = openmc.data.dose_coefficients('photon', geometry=coeff_geom)
+    dose_filter = openmc.EnergyFunctionFilter(dose_energy, dose)
+
+    # An energy filter is created to assign to the flux tally.
+    energy_filter_flux = openmc.EnergyFilter.from_group_structure("VITAMIN-J-42")
+
+    spectrum_tally = openmc.Tally(tally_id=2, name="Flux spectrum")
+    # Implementing energy and cell filters for flux spectrum tally
+    spectrum_tally.filters = [cell_filter, total_filter, energy_filter_flux, particle_filter, dose_filter]

```

Comment:
maybe you should apply the `dose_filter` to the `photon_tally` instead of the `spectrum_tally`?